src/types/form_logic.ts

```diff
@@ -51,3 +52,55 @@ export interface IPreventSubmitLogic extends ILogic {
 export interface IPreventSubmitLogicSchema
   extends IPreventSubmitLogic,
     Document {}
+
+type LogicField = Extract<
+  BasicField,
+  | BasicField.Dropdown
+  | BasicField.Radio
+  | BasicField.YesNo
+  | BasicField.Number
+  | BasicField.Decimal
+  | BasicField.Rating
+>
```

Comment:
`Extract` doesn't really help us here, right? is this simpler?
```suggestion
type LogicField =
  | BasicField.Dropdown
  | BasicField.Radio
  | BasicField.YesNo
  | BasicField.Number
  | BasicField.Decimal
  | BasicField.Rating
```